aspnet/fundamentals/file-providers.rst

```diff
@@ -1,8 +1,149 @@
+^^^^^^^^^^^^^^^^^^^^
+
+The ``EmbeddedFileProvider`` is used to access files embedded in assemblies. In .NET Core, you embed files in an assembly by specifying them in ``buildOptions`` in the *project.json* file:
+
+.. literalinclude:: file-providers/sample/src/FileProviderSample/project.json
+  :lines: 42-49
+  :emphasize-lines: 4-7
+
+You can use `globbing patterns`_ when specifying files to embed in the assembly.
+
+When creating an ``EmbeddedFileProvider``, pass the assembly it will read to its constructor.
+
+.. code-block:: c#
+
+  var embeddedProvider = new EmbeddedFileProvider(Assembly.GetEntryAssembly());
+
+The snippet above demonstrates how to create an ``EmbeddedFileProvider`` with access to the currently executing assembly.
+
+Updating the sample app to use an ``EmbeddedFileProvider`` results in the following output:
+
+.. image:: /fundamentals/file-providers/_static/embedded-directory-listing.png
+
+.. note:: Embedded resources do not expose directories. Rather, the path to the resource is embedded in its filename using ``.`` separators.
+
+CompositeFileProvider
+^^^^^^^^^^^^^^^^^^^^^
+
+The ``CompositeFileProvider`` combines ``IFileProvider`` instances, exposing a single interface for working with files from multiple providers. When creating the ``CompositeFileProvider``, you pass one or more ``IFileProvider`` instances to its constructor:
+
+.. literalinclude:: file-providers/sample/src/FileProviderSample/Startup.cs
+  :lines: 34-36
+  :emphasize-lines: 3
+  :dedent: 12
+
+Updating the sample app to use a ``CompositeFileProvider`` that includes both the physical and embedded providers configured previously, results in the following output:
+
+.. image:: /fundamentals/file-providers/_static/composite-directory-listing.png
+
+
+Watching for changes
+--------------------
+
+The ``IFileProvider`` ``Watch`` method provides a way to watch one or more files or directories for changes. This method accepts a path string, which can use `globbing patterns`_ to specify multiple files, and returns an ``IChangeToken``. This token exposes a ``HasChanged`` property that can be inspected, and a ``RegisterChangeCallback`` method that is called when changes are detected to the specified path string.
+
+In this article's sample, the ``Watch`` action on the ``HomeController`` sets up a watch on a particular file, and configures a callback that sets a static field on the controller with the time when the file was last changed. Note that each change token will only call its callback once - if you need to constantly monitor for changes, you will need create a new token each time a change is fired.
+
+.. literalinclude:: file-providers/sample/src/FileProviderSample/Controllers/HomeController.cs
+  :lines: 23-33
+  :emphasize-lines: 3-7
+  :dedent: 8
+
+This example will register a new callback every time a request is handled by the ``Watch`` action. 
+**REVIEWER**: Is there a better example for using Watch in an MVC action that would avoid this behavior?
```

Comment:
@danroth27 Open to ideas for how to improve this bit of the sample (or let me know if you think it's fine as-is).